vernac/retrieveObl.ml

```diff
@@ -203,12 +203,17 @@ let sort_dependencies evl =
   in
   aux evl Evar.Set.empty []
 
+let collect_evars_of_term evd c ty =
+  Evar.Set.union (Evd.evars_of_term evd c) (Evd.evars_of_term evd ty)
+
 let retrieve_obligations env name evm fs ?status t ty =
   (* 'Serialize' the evars *)
   let nc = Environ.named_context env in
   let nc_len = Context.Named.length nc in
   let evm = Evarutil.nf_evar_map_undefined evm in
   let evl = Evarutil.non_instantiated evm in
```

Comment:
unrelated to this pr but non_instantiated also does nf so there is redundancy here